### ISSUE TITLE
ESP32 DEVKIT V4 issues with sketch

### ISSUE THREAD

**Issue by Rauf56** [platformio/platform-espressif32#1484]
![IMG_6951](https://github.com/user-attachments/assets/4931e5cf-666b-49b7-a7dd-5b04443d9464)

Hello everyone, I'm new to GitHub, and I'm not sure if I'm posting this in the right place. I'm having an issue with my ESP32 DEVKIT v4 board, specifically with a sketch in the Arduino IDE. My program receives data via Serial Transfer, specifically: `valueMotor`, `valueAngle`, and `valueZnaki`, as can be seen when data is received. 

The problem occurs when a value is assigned to the `valueZnaki` variable. On the original Arduino ESP32 board, this sketch worked correctly, but on this board, it starts to freeze whenever the program enters any of the conditions involving `valueZnaki` and doesn’t exit them. I'm not sure what’s causing this issue. I tried troubleshooting, but I couldn’t figure it out. Could someone please help?

**Comment by Rauf56** [comment]
![IMG_6952](https://github.com/user-attachments/assets/034d5b46-07d7-4628-b217-6ff83c48905c)

configuration

**Comment by Rauf56** [comment]
After a piece of torn hair from my bald head and 3 hours of attempts and meetings with my comrades, it turned out that after all, the Chinese board does not read data from the buffer as quickly as the original one, which could be influenced by a bunch of factors. After all, a lot of time has passed since I switched from the Arduino board to the Chinese copy and my code has changed a lot in size. So, the ‘moments’ that turned out to be problematic: 1. ‘Buffer overload’ with data (data was sent from python without any delay). I added a delay of about 150 milliseconds in sending data from python, after which the program began to hang, and not to freeze as before. 2. ‘Add a separate counter’ to each individual condition, which will be replenished if the condition has already been entered. And I do not know how to explain another important point. If no characters were found in python, then zero was sent to the variable ’valueZnaki’ according to the series, after which all counters were reset to zero.I do not know why just the flags did not work hahaha.  I've learned a lot from this today. After all, I had been struggling with this for 2 more days before that

**Comment by Rauf56** [comment]
After a piece of torn hair from my bald head and 3 hours of attempts and meetings with my comrades, it turned out that after all, the Chinese board does not read data from the buffer as quickly as the original one, which could be influenced by a bunch of factors. After all, a lot of time has passed since I switched from the Arduino board to the Chinese copy and my code has changed a lot in size. So, the `moments' that turned out to be problematic: 1. `Buffer overload' with data (data was sent from python without any delay). I added a delay of about 150 milliseconds in sending data from python, after which the program began to hang, and not to freeze tightly as before. 2. `Add a separate counter' to each individual condition, which will be replenished if the condition has already been entered. And I do not know how to explain another important point. If no characters were found in python, then zero was sent to the variable `Valueznaki' according to the series, after which all counters were reset to zero.I do not know why just the flags did not work hahaha.  I've learned a lot from this today. After all, I had been struggling with this for 2 more days before that
